pkg/core/region_tree.go

```diff
@@ -438,8 +438,8 @@ func (t *regionTree) RandomRegions(n int, ranges []keyutil.KeyRange) []*RegionIn
 		}
 	)
 	// This is a fast path to reduce the unnecessary iterations when we only have one range.
-	if rangeLen <= 1 {
-		if rangeLen == 1 {
+	if len(ranges) <= 1 {
+		if len(ranges) == 1 {
```

Comment:
Reason for this change?

Reply:
<img width="742" height="103" alt="image" src="https://github.com/user-attachments/assets/6ddc7cc1-fe75-4fd0-b1cf-04471a1426ea" />

😂